src/components/TDViewer/Form.jsx

```diff
@@ -0,0 +1,160 @@
+                <div className="text-formBlue place-self-center text-center text-xs px-4">Write</div>
+            </div>
+            <div className=" place-self-center pl-3 text-base text-white overflow-hidden flex-grow">{props.form.href}</div>
+            <button className="text-base w-6 h-6 p-1 shadow-md rounded-full bg-formBlue" onClick={() => deleteForm(props)}>
+                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor">
+                    <path fillRule="evenodd" d="M9 2a1 1 0 00-.894.553L7.382 4H4a1 1 0 000 2v10a2 2 0 002 2h8a2 2 0 002-2V6a1 1 0 100-2h-3.382l-.724-1.447A1 1 0 0011 2H9zM7 8a1 1 0 012 0v6a1 1 0 11-2 0V8zm5-1a1 1 0 00-1 1v6a1 1 0 102 0V8a1 1 0 00-1-1z" clipRule="evenodd" />
+                </svg>
+            </button>
+        </div>
+    );
+}
+
+export function InvokeForm(props) {
+    const context = useContext(ediTDorContext)
+    const deleteForm = (e) => {
+        context.removeForm(e)
+    }
+    return (
+        <div className="flex flex-row items-center justify-start h-10 w-full bg-formRed rounded-md px-4 mt-2 bg-opacity-75 border-2 border-formRed">
+            <div className="flex h-6 w-16 bg-white rounded-md place-self-center justify-center">
+                <div className="text-formRed place-self-center text-center text-xs px-4">Invoke</div>
+            </div>
+            <div className=" place-self-center pl-3 text-base text-white overflow-hidden flex-grow">{props.form.href}</div>
+            <button className="text-base w-6 h-6 p-1 shadow-md rounded-full bg-formRed" onClick={() => deleteForm(props)}>
+                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor">
+                    <path fillRule="evenodd" d="M9 2a1 1 0 00-.894.553L7.382 4H4a1 1 0 000 2v10a2 2 0 002 2h8a2 2 0 002-2V6a1 1 0 100-2h-3.382l-.724-1.447A1 1 0 0011 2H9zM7 8a1 1 0 012 0v6a1 1 0 11-2 0V8zm5-1a1 1 0 00-1 1v6a1 1 0 102 0V8a1 1 0 00-1-1z" clipRule="evenodd" />
+                </svg>
+            </button>
+        </div>
+    );
+}
+
+export function UndefinedForm(props) {
+    const context = useContext(ediTDorContext)
+    const deleteForm = (e) => {
+        context.removeForm(e)
+    }
+    return (
+        <div className="flex flex-row items-center justify-start h-10 w-full bg-gray-300 rounded-md px-4 mt-2 bg-opacity-75 border-2 border-gray-300">
+            <div className="flex h-6 w-16 bg-white rounded-md place-self-center justify-center">
+                <div className="place-self-center text-center text-xs px-4">Undefined</div>
+            </div>
+            <div className=" place-self-center pl-3 text-base overflow-hidden">{props.form.href}</div>
+            <button className="text-base w-6 h-6 p-1 shadow-md rounded-full bg-white" onClick={() => deleteForm(props)}>
+                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor">
```

Comment:
Also what do you think about making the delete button a more prominent and "scary" by giving the icon e.g. a red color on a white background or something?

Just to make sure no user presses the button without a second thought and regrets it immediately after.

Reply:
I'm not sure that this is a good idea, because the user should not be scared when working with a TD and I also think deleting a Form is a normal step when working with a TD.